src/app/event.service.ts

```diff
@@ -15,35 +16,89 @@ export class EventService implements OnDestroy {
   public constructor(
     private socketService: SocketService,
     private configService: ConfigService,
+    private notificationService: NotificationService,
     private router: Router,
   ) {
     this.subscriptions.add(
-      this.socketService.getEventSubscribable().subscribe((event: PrinterEvent) => {
-        if (event === PrinterEvent.PRINTING || event === PrinterEvent.PAUSED) {
-          setTimeout(() => {
-            this.printing = true;
-          }, 500);
+      this.socketService.getEventSubscribable().subscribe((event: PrinterEvent | PrinterNotification) => {
+        if (this.isPrinterNotification(event)) {
+          this.handlePrinterNotification(event);
         } else {
-          setTimeout(() => {
-            this.printing = false;
-          }, 1000);
-        }
-
-        if (event === PrinterEvent.CLOSED) {
-          this.router.navigate(['/standby']);
-        } else if (event === PrinterEvent.CONNECTED) {
-          setTimeout(() => {
-            if (this.configService.isTouchscreen()) {
-              this.router.navigate(['/main-screen']);
-            } else {
-              this.router.navigate(['/main-screen-no-touch']);
-            }
-          }, 1000);
+          this.handlePrinterEvent(event);
         }
       }),
     );
   }
 
+  // https://stackoverflow.com/questions/14425568/interface-type-check-with-typescript
+  // https://stackoverflow.com/questions/8511281/check-if-a-value-is-an-object-in-javascript/8511350#8511350
+  private isPrinterNotification(object: any): object is PrinterNotification {
```

Comment:
Maybe another way of doing this would be to check whether the received message is NOT of type `int` (which would indicate that it's an enum, thus PrinterEvent). It should work (haven't tested it though), but might be a bit cleaner.

I wouldn't switch the logic around though, as the default should be handle PrinterEvent. Alternatively you could check whether the received message is of type `object`.